aicsimageio/exceptions.py

```diff
@@ -10,19 +10,25 @@ class ConflictingArgumentsError(Exception):
     pass
 
 
+class InvalidDimensionOrderingError(Exception):
+    """
+    A general exception that can be thrown when handling dimension ordering or
+    validation. Should be provied a message for the user to be given more context.
+    """
+
+    pass
+
+
 class UnsupportedFileFormatError(Exception):
     """
     This exception is intended to communicate that the file extension is not one of
     the supported file types and cannot be parsed with AICSImage.
     """
 
-    def __init__(self, reader_name: str, extension: str, **kwargs):
+    def __init__(self, reader_name: str, path: str, **kwargs):
         super().__init__(**kwargs)
         self.reader_name = reader_name
-        self.extension = extension
+        self.path = path
 
     def __str__(self):
-        return (
-            f"{self.reader_name} does not support the "
-            f"image file type: '{self.extension}'."
-        )
+        return f"{self.reader_name} does not support the image: '{self.path}'."
```

Comment:
Allows better handling for things like ome tiff files which have multiple extensions + supporting fake paths through passing the type of the unsupported object.